server/src/main/scala/actors/CrunchStateActor.scala

```diff
@@ -115,9 +115,13 @@ class CrunchStateActor(initialMaybeSnapshotInterval: Option[Int],
       log.error(s"Received unexpected message $u")
   }
 
-  def stateForPeriod(start: MillisSinceEpoch, end: MillisSinceEpoch): Option[PortState] = {
-    logDebug(s"PortState contains: (cms, fs, sms) ${state.map(s => (s.crunchMinutes.size, s.flights.size, s.staffMinutes.size).toString()).getOrElse("Nothing")}")
-    state.map(_.window(SDate(start), SDate(end), portQueues))
+  def stateForPeriod(start: MillisSinceEpoch, end: MillisSinceEpoch, maybeTerminalName: Option[TerminalName]): Option[PortState] = {
+    maybeTerminalName match {
```

Comment:
break this into a separate function to filter by terminal.

Reply:
Done